exercises/00-init/src/modules/root/components/root.tsx

```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 
-export const Root = () => (
+export const Root: React.FC = () => (
```

Comment:
I've decided to add this type check for multiple reasons:
1. consistency with other instances
2. my belief that it's better to be more explicit (at least in training), then less explicit. It's better to know that something exists and not use it because it seems unnecessary, then not using it because of not knowing. 
3. this type check additionally checks the return type, which seems to produce better errors in my opinion:
    - with `React.FC` it points to the return value and prints out
      ```
      Type 'number' is not assignable to type 'ReactElement<any, any> | null'
      ```
      The potential error could be found much faster
    - without `React.FC` it points to the place where the component is used and prints out
      ```
      'Root' cannot be used as a JSX component.
      Its return type 'number' is not a valid JSX element.
      ```
      Not checking the return type seemed to me like not using the full potential of the TypeScript.

It seems, that I'm outnumbered here, so if you confirm that you want it gone, I'll delete it

Reply:
I agree with your reasoning, and would prefer to see the explicit typing remain.

Reply:
Just to clarify, I don't think this is wrong or bad or anything like that - it's just useless in my opinion, so I have to disagree with advantages of your arguments. :innocent: 

1. Consistency. I think this is more about explicit vs implicit (inferred) typing. I would definitely vote for the second one. Do we want to **always** (to be consistent) use `const abc: number = 42;`? I hope we don't :slightly_smiling_face: because it's enough to use `const abc = 42;`.
2. Explicit typing. The only "benefit" of explicit typing I see is that you will see the type in GitHub. I mean, for a component like this, it's still useless because it's obvious. For other things, it might be beneficial but not enough (for me) to write explicit typing everywhere.
3. Full potential of TS. Well, actually, we don't use the full potential of TS with explicit typing because we don't use type inference. Honestly, even the second error message is great IMHO and it helps to find out what the problem is.

If we want to keep it for teaching purposes like _"Hey, there's this type `React.FC` which **can** be used for components."_ then it's absolutely OK but I don't think it's good to force newcomers to write explicit typing when it's not necessary - like in this case. :slightly_smiling_face:

Reply:
FWIW, I disagree with your response to point 1 @branoholy .
This is more equivalent to something like `const abc: number = '42';`, if such a thing was legal in typescript, because by giving the variable an explicit type, you're actually narrowing it, which is why you get the different error Dina mentioned.

So it's not about implicit vs explicit typing at all IMO, it's absolutely about establishing consistency. I.e. that all React components must be of the type `React.FC`, and not one of the other allowed values. I'm not sure that is always the case, but IMO it doesn't hurt to establish that as the rule, and consciously make an exception where needed.

Reply:
> This is more equivalent to something like `const abc: number = '42';`, if such a thing was legal in typescript, because by giving the variable an explicit type, you're actually narrowing it, which is why you get the different error Dina mentioned.

Yeah, this is an example when the value is wrong. Mine are examples when the values are correct.

> So it's not about implicit vs explicit typing at all IMO, it's absolutely about establishing consistency. I.e. that all React components must be of the type React.FC, and not one of the other allowed values. I'm not sure that is always the case, but IMO it doesn't hurt to establish that as the rule, and consciously make an exception where needed.

Sorry, I think we are matching different patterns to this. You see "consistency for React components", I see "explicit typing that is not necessary". I would leave it up to @novellizator to decide.

Just a side note: The typing of `React.FC` is not correct. A function that returns a number is a valid React component.

Reply:
Leaving as it is